experimental/generation/generator/packages/library/test/generate.test.ts

```diff
@@ -154,6 +158,43 @@ describe('dialog:generate library', async () => {
         }
     })
 
+    it('LU and LG Unit Test', async () => {
+        for (let i = 1; i <= 23; i++) {
+            await gen.generate(`${unitTestSchemaPath}${i}.schema`, undefined, output, undefined, ['en-us'], undefined, false, false, true, feedback)
+            try {
+                console.log(`\n\nLG Testing schema ${i}`)
+                const templates = Templates.parseFile(`${output}/en-us/unittest${i}.en-us.lg`)
+                const allDiagnostics = templates.allDiagnostics
+                if (allDiagnostics) {
+                    let errors = allDiagnostics.filter((u): boolean => u.severity === DiagnosticSeverity.Error)
+                    if (errors && errors.length > 0) { throw new Error('LG format error') }
+                }
+            } catch (e) {
+                assert(e.message)
+            }
+            try {
+                console.log(`\n\nLG Testing schema ${i}`)
+                let result: any
+                const luFiles = await file.getLuObjects(undefined, `${output}/en-us/unittest${i}.en-us.lu`, true, ".lu")
+                result = await LuisBuilder.build(luFiles, true, "en-us")
+                result.validate()
+                if (!hasContent(result)) {
+                    throw new Error('LU format error')
```

Comment:
>'LU format error' [](start = 36, length = 17)

Surface error

Reply:
Surface the actual errors.  You might want to use the assertion mechanism though rather than throwing.

---
In reply to: [513126741](https://github.com/microsoft/BotBuilder-Samples/pull/2854#discussion_r513126741) [](ancestors = 513126741)